src/data/polynomial/taylor.lean

```diff
@@ -92,4 +137,37 @@ begin
   simp only [taylor_coeff, h, coeff_zero],
 end
 
+lemma sum_taylor_X_pow {R} [comm_ring R] (r : R) (n : ℕ) :
+  (taylor r (X ^ n)).sum (λ i a, C a * (X - C r) ^ i) = X ^ n :=
+begin
+  nontriviality R,
+  rw sum_over_range,
+  { rw nat_degree_taylor,
+    nth_rewrite_rhs 0 ←sub_add_cancel X (C r),
+    rw add_pow,
+    refine finset.sum_congr _ _,
+    { simp },
+    { intros x hx,
+      rw taylor_coeff,
+      simp [X_pow_eq_monomial, mul_comm, mul_assoc, mul_left_comm] } },
+  { simp }
+end
+
+/-- Taylor's formula. -/
+lemma sum_taylor_eq {R} [comm_ring R] (f : polynomial R) (r : R) :
```

Comment:
Does it make sense to first prove `taylor r f = f.sum blabla`? And then `taylor r (taylor s f) = `taylor (r + s) f`?
(Both of these probably need some `comm_ring`-like assumptions.)
Because this lemma is basically `taylor (-r) (taylor r f) = f`, right?